ext/AbstractDifferentiationForwardDiffExt.jl

```diff
@@ -61,20 +61,45 @@ function AD.hessian(ba::AD.ForwardDiffBackend, f, x::AbstractArray)
     return (ForwardDiff.hessian(f, x, cfg),)
 end
 
+function AD.value_and_derivative(::AD.ForwardDiffBackend, f, x::Real)
+    T = typeof(ForwardDiff.Tag(f, typeof(x)))
+    ydual = f(ForwardDiff.Dual{T}(x, one(x)))
+    return ForwardDiff.value(T, ydual), (ForwardDiff.partials(T, ydual, 1),)
+end
+
 function AD.value_and_gradient(ba::AD.ForwardDiffBackend, f, x::AbstractArray)
     result = DiffResults.GradientResult(x)
     cfg = ForwardDiff.GradientConfig(f, x, chunk(ba, x))
     ForwardDiff.gradient!(result, f, x, cfg)
     return DiffResults.value(result), (DiffResults.derivative(result),)
 end
 
+function AD.value_and_second_derivative(ba::AD.ForwardDiffBackend, f, x::Real)
+    T = typeof(ForwardDiff.Tag(f, typeof(x)))
+    xdual = ForwardDiff.Dual{T}(x, one(x))
+    T2 = typeof(ForwardDiff.Tag(f, typeof(xdual)))
```

Comment:
can you explain how this works? tags are not documented anywhere

Reply:
TBH I'm cargo-culting this from `ForwardDiff`:

https://github.com/JuliaDiff/ForwardDiff.jl/blob/5e9240ae63e5d4217e1dc17cc14deca1d74c1ad7/src/derivative.jl#L12-L15

Tagging is there to [avoid perturbation confusion](https://juliadiff.org/ForwardDiff.jl/stable/user/advanced/#Custom-tags-and-tag-checking) in the presence of nested differentiation calls. Indeed there's no additional documentation and unfortunately that's pretty much the extent to which I'm familiar with this functionality.

In `ForwardDiff.derivative`, the `Tag` type is created from the function and the type of the input; so I'm copying exactly that behavior for higher-order differentiation too...

Reply:
But generally you may want other tags (e.g., SciML and Turing use custom tags), see https://www.stochasticlifestyle.com/improved-forwarddiff-jl-stacktraces-with-package-tags/

Reply:
I think @devmotion might be more qualified than us to approve of this part

Reply:
@devmotion can we merge this and possibly discuss custom tags in a different PR?